shared/src/paraswap_api.rs

```diff
@@ -269,31 +271,56 @@ pub struct TransactionBuilderQuery {
     pub src_token: H160,
     /// The received token
     pub dest_token: H160,
-    /// The source amount
-    #[serde(with = "u256_decimal")]
-    pub src_amount: U256,
-    /// The amount (from priceRoute) - slippage
-    #[serde(with = "u256_decimal")]
-    pub dest_amount: U256,
+    /// The trade amount amount
+    #[serde(flatten)]
+    pub trade_amount: TradeAmount,
     /// The decimals of the source token
-    pub from_decimals: usize,
+    pub src_decimals: usize,
     /// The decimals of the destination token
-    pub to_decimals: usize,
+    pub dest_decimals: usize,
     /// priceRoute part from /prices endpoint response (without any change)
     pub price_route: Value,
     /// The address of the signer
     pub user_address: H160,
-    /// partner's referrer string, important if the partner takes fees
-    pub referrer: String,
 }
 
-impl TransactionBuilderQuery {
+/// The amounts for buying and selling.
+#[derive(Clone, Debug, Eq, PartialEq, Serialize)]
+#[serde(untagged)]
+pub enum TradeAmount {
+    #[serde(rename_all = "camelCase")]
+    Sell {
+        /// The source amount
+        #[serde(with = "u256_decimal")]
+        src_amount: U256,
+        /// The maximum slippage in BPS.
+        slippage: u32,
+    },
+    #[serde(rename_all = "camelCase")]
+    Buy {
+        /// The destination amount
+        #[serde(with = "u256_decimal")]
+        dest_amount: U256,
+        /// The maximum slippage in BPS.
+        slippage: u32,
+    },
+}
+
+#[derive(Clone, Debug, Serialize)]
+#[serde(rename_all = "camelCase")]
+struct TransactionBuilderQueryWithPartner<'a> {
+    #[serde(flatten)]
+    query: TransactionBuilderQuery,
+    partner: &'a str,
```

Comment:
This seems to be set but never read. Probably not intentional.

Reply:
Its included in the JSON that we POST to the ParaSwap API.

Reply:
Ah so its  part of the url for prices but body for transaction. Makes sense.